Survey/Sources/Presentation/Modules/Home/HomeView.swift

```diff
@@ -36,82 +36,129 @@ struct HomeView: View {
+        let surveyList = Array(output.surveys.enumerated())
+
+        return TabView(selection: $tabSelection) {
+            ForEach(surveyList, id: \.element.id) { args in
                 let (index, survey) = args
-                SurveyItemView(
-                    survey: survey,
-                    willGoToDetail: {
-                        withoutAnimation {
-                            isModalPresented.toggle()
-                        }
-                    }
+                surveyItemViewSetup(
+                    with: index,
+                    and: survey
                 )
-                .tag(index)
-                .onAppear {
-                    currentPage = index
-                }
-                .fullScreenCover(isPresented: $isModalPresented) {
-                    SurveyDetailView(
-                        viewModel: SurveyDetailViewModel(),
-                        isPresented: $isModalPresented,
-                        survey: survey
-                    )
-                }
             }
             .highPriorityGesture(DragGesture().onEnded {
                 handleSwipe(translation: $0.translation.width)
             })
         }
         .tabViewStyle(PageTabViewStyle(indexDisplayMode: .never))
         .overlay(alignment: .leading) {
-            VStack(alignment: .leading) {
-                Spacer()
-                PageControlView(
-                    currentPage: $currentPage,
-                    numberOfPages: list.count,
-                    didSelect: { index in
-                        tabSelection = index
-                    }
-                )
-                .frame(
-                    width: .zero,
-                    height: 8.0,
-                    alignment: .leading
-                )
-                .accessibilityIdentifier(TestConstants.Home.pageIndicator)
-                .padding(.bottom, 200.0)
-            }
+            pageControlViewSetup()
+        }
+        .alert(isPresented: .constant($output.alert.wrappedValue != nil)) {
+            Alert(
+                title: Text(output.alert?.title ?? ""),
+                message: Text(output.alert?.message ?? ""),
+                dismissButton: .default(Text("OK"), action: {
+                    $output.alert.wrappedValue = nil
+                })
+            )
         }
         .edgesIgnoringSafeArea(.all)
     }
 
+    private func surveyItemViewSetup(with index: Int, and survey: Survey) -> some View {
+        SurveyItemView(
+            survey: survey,
+            willGoToDetail: {
+                willGoToDetail.send()
+            }
+        )
+        .tag(index)
+        .onChange(of: tabSelection, perform: { newValue in
+            currentPage = newValue
+        })
+        .onReceive(output.$willGoToDetail) {
+            guard $0 else { return }
+            withoutAnimation {
+                isModalPresented = true
+            }
+        }
+        .onAppear {
+            isModalPresented = false
```

Comment:
what happen if we don't have this line?

Reply:
Nice catch 🙌 . If we have that line or not, nothing special happens 😄 . I removed it at https://github.com/nkhanh44/ic-survey-ios/pull/52/commits/980a8a5e249ebb8cf5b480d1ec1c132b82f6bebc